terraform/hcl.go

```diff
@@ -0,0 +1,248 @@
+package terraform
+
+import (
+	"fmt"
+	"path"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/hashicorp/terraform/configs"
+	"github.com/spf13/afero"
+	"github.com/zclconf/go-cty/cty"
+
+	"github.com/cycloidio/terracost/query"
+)
+
+// ExtractQueriesFromHCL returns the resources found in the module identified by the modPath.
+func ExtractQueriesFromHCL(fs afero.Fs, providerInitializers []ProviderInitializer, modPath string) ([]query.Resource, error) {
+	parser := configs.NewParser(fs)
+	mod, diags := parser.LoadConfigDir(modPath)
+	if diags.HasErrors() {
+		return nil, fmt.Errorf(diags.Error())
+	}
+
+	evalCtx := getEvalCtx(mod, nil)
+
+	providers, err := getHCLProviders(mod, evalCtx, providerInitializers)
+	if err != nil {
+		return nil, err
+	}
+
+	return extractHCLModule(providers, parser, modPath, "", mod, evalCtx)
+}
+
+// extractHCLModule returns the resources found in the provided module.
+func extractHCLModule(providers map[string]Provider, parser *configs.Parser, modPath string, modName string, mod *configs.Module, evalCtx *hcl.EvalContext) ([]query.Resource, error) {
+	queries := make([]query.Resource, 0, len(mod.ManagedResources))
+
+	for rk, rv := range mod.ManagedResources {
+		if modName != "" {
+			rk = fmt.Sprintf("module.%s.%s", modName, rk)
+		}
+
+		providerKey := rv.Provider.Type
+		if rv.ProviderConfigRef != nil {
+			providerKey = rv.ProviderConfigRef.String()
+		}
+		provider := providers[providerKey]
```

Comment:
Can only be found I guess?

Reply:
Since it's an `interface{}`, it will be `nil` if not found in the map. This is then checked later down: https://github.com/cycloidio/terracost/pull/45/files/5583ecf7da89196e5d59e1c64f1d5838d13148e3..4e0d5949fce9e0218d1f2b9f0f9184dc967e4952#diff-d88d3b92409f1211e4b636160025511bac9b8ab47d7c06375fd0092f53cf5763R69